### PR TITLE
Docker and Fabrication

### PR DESCRIPTION
**Pull request by ShubhamSureshPatil**
I have added two new files under fabrication and tools. Please do a review soon. Thanks

### CODE REVIEW

**Comment by benbo** [comment]
Thanks Shubham, these look like solid contributions. I think you could drop the reference to the MRSD printer since the aim is for this to be a general purpose robotics knowledgeable.

**Comment by ShubhamSureshPatil** [comment]
Hi Ben and Bryan. I have removed the MRSD printer reference and replaced it with the one available in the market.

**Comment by ShubhamSureshPatil** [comment]
Hi,

I have made the changes as requested. Can you please look into it?

Thanks

File: wiki/fabrication/fabrication_considerations_for_3D_printing.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+- Print bed temperature: 80°C – 110°C
+- Shrinkage/ warping: Considerable
+- Soluble: In esters, ketones, and acetone.
+
+   [create_docker_image]: <https://www.mirantis.com/blog/how-do-i-create-a-new-docker-image-for-my-application/>
+   [install_nvidia_driver]: <https://github.com/heethesh/Computer-Vision-and-Deep-Learning-Setup>
+
+#### Furthur Reading:
+1. Design considerations for 3D printing.
+<https://www.3dhubs.com/knowledge-base/key-design-considerations-3d-printing>
```
Comment (smbryan) [inline wiki/fabrication/fabrication_considerations_for_3D_printing.md]: Wrong format. Should be [title](link)

File: wiki/fabrication/fabrication_considerations_for_3D_printing.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
+# Fabrication Considerations for 3D printing
```
Comment (smbryan) [inline wiki/fabrication/fabrication_considerations_for_3D_printing.md]: Should be
```
---
title: Fabrication Considerations for 3D Printing
---
```

File: wiki/fabrication/fabrication_considerations_for_3D_printing.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+ABS properties:
+- Strength: High | Flexibility: Medium | Durability: High
+- Difficulty to use: Medium
+- Print temperature: 210°C – 250°C
+- Print bed temperature: 80°C – 110°C
+- Shrinkage/ warping: Considerable
+- Soluble: In esters, ketones, and acetone.
+
+   [create_docker_image]: <https://www.mirantis.com/blog/how-do-i-create-a-new-docker-image-for-my-application/>
```
Comment (smbryan) [inline wiki/fabrication/fabrication_considerations_for_3D_printing.md]: Wrong format. Should be `title`

File: wiki/tools/docker.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    (amd64)
+ 3. The Docker daemon created a new container from that image which runs the
+    executable that produces the output you are currently reading.
+ 4. The Docker daemon streamed that output to the Docker client, which sent it
+    to your terminal.
+```
+
+To Create docker files for your project, you can follow the tuturial [here][create_docker_image]
+
+   [create_docker_image]: <https://www.mirantis.com/blog/how-do-i-create-a-new-docker-image-for-my-application/>
```
Comment (smbryan) [inline wiki/tools/docker.md]: Format.

**Comment by smbryan** [comment]
@ShubhamSureshPatil I've added a few things to clean up. Looking forward to getting this one in!

File: wiki/tools/docker.md [hunk 5]
```diff
@@ -0,0 +1,172 @@
+# Docker Setup
```
Comment (smbryan) [inline wiki/tools/docker.md]: Title missing.

File: wiki/fabrication/fabrication_considerations_for_3D_printing.md (hunk 2 above)
Comment (ShubhamSureshPatil) [reply wiki/fabrication/fabrication_considerations_for_3D_printing.md]: Incorporated the change

File: wiki/fabrication/fabrication_considerations_for_3D_printing.md (hunk 1 above)
Comment (ShubhamSureshPatil) [reply wiki/fabrication/fabrication_considerations_for_3D_printing.md]: Correct the format for the entire '_Furthur Reading_' section.

File: wiki/fabrication/fabrication_considerations_for_3D_printing.md (hunk 3 above)
Comment (ShubhamSureshPatil) [reply wiki/fabrication/fabrication_considerations_for_3D_printing.md]: Updated to as per the correct format

File: wiki/tools/docker.md (hunk 5 above)
Comment (ShubhamSureshPatil) [reply wiki/tools/docker.md]: Added the title 'Docker'

File: wiki/tools/docker.md (hunk 4 above)
Comment (ShubhamSureshPatil) [reply wiki/tools/docker.md]: Updated the format.

**Comment by ShubhamSureshPatil** [comment]
Hi Sean,

I have made the changes as requested. Please have a look. Thanks

File: wiki/tools/docker.md [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+ 4. The Docker daemon streamed that output to the Docker client, which sent it
+    to your terminal.
+```
+
+To Create docker files for your project, you can follow the tuturial [here](<https://www.mirantis.com/blog/how-do-i-create-a-new-docker-image-for-my-application/>)
+
+### Furthur Reading:
+1. [Create your very own Docker image](https://www.scalyr.com/blog/create-docker-image)
+
+2. [Create Docker containers, services,  swarms, stacks for your application](<https://docs.docker.com/get-started>)
```
Comment (smbryan) [inline wiki/tools/docker.md]: Need to drop the `<>`

File: wiki/tools/docker.md [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+
+3. [Deploy Docker containers in AWS](<https://aws.amazon.com/getting-started/tutorials/deploy-docker-containers>)
+
+4. [Docker Images inventory](<https://hub.docker.com/search/?type=image>)
+
+#### References:
+1. About Docker:
```
Comment (smbryan) [inline wiki/tools/docker.md]: These need to be updated too.

File: wiki/fabrication/fabrication_considerations_for_3D_printing.md [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+
+3. [Dimensionality test](<https://www.youmagine.com/designs/make-2016-3d-printer-shootout-models>)
+
+4. [How to fix warping: Tutorial bu Ultimaker](<https://ultimaker.com/en/resources/19537-how-to-fix-warping>)
+
+5. [Accuracy of 3D printing materials](<https://www.3dhubs.com/knowledge-base/dimensional-accuracy-3d-printed-parts>)
+
+#### References:
+1. Printing using ABS material:
+<https://all3dp.com/abs-3d-printer-filament-explained>
```
Comment (smbryan) [inline wiki/fabrication/fabrication_considerations_for_3D_printing.md]: These need to be updated too.

File: wiki/fabrication/fabrication_considerations_for_3D_printing.md [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+- Soluble: In esters, ketones, and acetone.
+
+   [create_docker_image]: <https://www.mirantis.com/blog/how-do-i-create-a-new-docker-image-for-my-application/>
+   [install_nvidia_driver]: <https://github.com/heethesh/Computer-Vision-and-Deep-Learning-Setup>
+
+#### Furthur Reading:
+1. [Design considerations for 3D printing](
+<https://www.3dhubs.com/knowledge-base/key-design-considerations-3d-printing>)
+
+1. [Calibrate your 3D printer](<https://all3dp.com/2/how-to-calibrate-a-3d-printer-simply-explained>)
```
Comment (smbryan) [inline wiki/fabrication/fabrication_considerations_for_3D_printing.md]: Two `1.`s in this list.

File: wiki/fabrication/fabrication_considerations_for_3D_printing.md [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+- Print bed temperature: 80°C – 110°C
+- Shrinkage/ warping: Considerable
+- Soluble: In esters, ketones, and acetone.
+
+   [create_docker_image]: <https://www.mirantis.com/blog/how-do-i-create-a-new-docker-image-for-my-application/>
+   [install_nvidia_driver]: <https://github.com/heethesh/Computer-Vision-and-Deep-Learning-Setup>
+
+#### Furthur Reading:
+1. [Design considerations for 3D printing](
+<https://www.3dhubs.com/knowledge-base/key-design-considerations-3d-printing>)
```
Comment (smbryan) [inline wiki/fabrication/fabrication_considerations_for_3D_printing.md]: Need to drop the `<>`.

File: wiki/fabrication/fabrication_considerations_for_3D_printing.md (hunk 3 above)
Comment (smbryan) [reply wiki/fabrication/fabrication_considerations_for_3D_printing.md]: Need to drop the `<>`.

File: wiki/fabrication/fabrication_considerations_for_3D_printing.md [hunk 11]
```diff
@@ -0,0 +1,155 @@
...
+
+The wall thickness defines the number of times the extruder will lay filament around the perimeter of your model before switching to your infill factor. (Infill is defined separately as a percentage. Typically, 10% or 20% works well for infill.)
+
+Your wall thickness should be defined as a multiple of your nozzle diameter, so a .4mm nozzle with a .8mm wall thickness would result in the printer laying a perimeter two thicknesses wide for each layer. This is the case with the Makerbot Replicator 2.
+
+It is most important to keep this in mind when you are drawing very thin walls. When parts of your model are thin, the perimeter walls are so close together that it doesn’t leave much room for infill in between them. Sometimes, it may not properly fuse the walls to each other, leaving a hollow gap between the walls on the inside of the model.
+
+
+## 2. Part Orientation
+![N|Solid](http://www.ap242.org/image/image_gallery?uuid=a7eb3359-8620-457d-9923-9d4cbec5ba5e&groupId=52520&t=1503324782232)
```
Comment (smbryan) [inline wiki/fabrication/fabrication_considerations_for_3D_printing.md]: Not sure about this, but we'll give it a shot.

**Comment by smbryan** [comment]
Few more issues. I'd recommend a spell check too while you're at it.

File: wiki/tools/docker.md (hunk 6 above)
Comment (ShubhamSureshPatil) [reply wiki/tools/docker.md]: updated by removing <>

File: wiki/tools/docker.md (hunk 7 above)
Comment (ShubhamSureshPatil) [reply wiki/tools/docker.md]: updated by removing <>

File: wiki/fabrication/fabrication_considerations_for_3D_printing.md (hunk 8 above)
Comment (ShubhamSureshPatil) [reply wiki/fabrication/fabrication_considerations_for_3D_printing.md]: removed <> and just kept the HTML links

File: wiki/fabrication/fabrication_considerations_for_3D_printing.md (hunk 9 above)
Comment (ShubhamSureshPatil) [reply wiki/fabrication/fabrication_considerations_for_3D_printing.md]: List order corrected

File: wiki/fabrication/fabrication_considerations_for_3D_printing.md (hunk 10 above)
Comment (ShubhamSureshPatil) [reply wiki/fabrication/fabrication_considerations_for_3D_printing.md]: Dropped the `<>`

File: wiki/fabrication/fabrication_considerations_for_3D_printing.md (hunk 11 above)
Comment (ShubhamSureshPatil) [reply wiki/fabrication/fabrication_considerations_for_3D_printing.md]: Tried putting the image into the asset section and then tried accessing it. Didn't work. Any other way of doing it?

**Comment by ShubhamSureshPatil** [comment]
Hi Sean,

I made the mentioned changes with spell checking. I tried putting the image
in the assets/image folder and then tried accessing it but it didn't work.
Anyway to resolve this?

**Comment by smbryan** [comment]
@ShubhamSureshPatil what you had would have worked from what I can tell, just not in raw form. You can [take a peak at the template.](https://raw.githubusercontent.com/RoboticsKnowledgebase/roboticsknowledgebase.github.io/master/_templates/template.md)

That being said, there's still several URLs that need to be formatted.

File: wiki/fabrication/fabrication_considerations_for_3D_printing.md [hunk 12]
```diff
@@ -16,7 +16,7 @@ It is most important to keep this in mind when you are drawing very thin walls.
 
 
 ## 2. Part Orientation
-![N|Solid](http://www.ap242.org/image/image_gallery?uuid=a7eb3359-8620-457d-9923-9d4cbec5ba5e&groupId=52520&t=1503324782232)
+![N|Solid](assets/images/part_orientation.jpeg)
```
Comment (smbryan) [inline wiki/fabrication/fabrication_considerations_for_3D_printing.md]: If you view this in "rich text" it renders correctly. However, the image description belongs in the `[]`.

Comment (ShubhamSureshPatil) [reply wiki/fabrication/fabrication_considerations_for_3D_printing.md]: Resolved

**Comment by ShubhamSureshPatil** [comment]
Hi Sean, Can you please review. I have kept a uniform format for URLs

**Comment by smbryan** [comment]
@ShubhamSureshPatil, looks good. @benbo I'm merging this one in.

**Pull request merged**